site/content/dev-guide/custom_http_adapter.md

```diff
@@ -0,0 +1,109 @@
++++
+title = "Implement a Custom Hono HTTP Protocol Adapter"
+weight = 100
++++
+
+Hono comes with the Hono REST protocol adapter which you may use as a blueprint for your own HTTP protocol adapter 
+implementation. 
+<!--more-->
+
+
+
+This section will guide you through the steps to build your own custom Hono HTTP protocol adapter.
+
+## Prerequisites
+
+You should be familiar with the setup and start of Hono. Refer to the 
+[Getting Started]({{< relref "getting-started.md" >}}) guide.
+
+## The Hono REST Protocol Adapter
+
+The Hono REST protocol adapter supports telemetry and event data processing. Additionally, it provides an interface for 
+device registration. 
+
+For documentation of the Hono REST protocol adapter, refer to 
+[REST Adapter]({{< relref "component/rest-adapter.md" >}}) 
+
+You can find the source of the Hono REST protocol adapter at 
+<https://github.com/eclipse/hono/tree/master/adapters/rest-vertx>.
+
+## Anatomy of the Hono REST Protocol Adapter
+ 
+Like many other Hono components, the Hono REST protocol adapter makes use of the [Vert.x](https://vertx.io) tool-kit. 
+All Hono protocol adapters are implemented as Vert.x verticles.
+
+The Hono REST protocol adapter class `VertxBasedRestProtocolAdapter` is derived from an abstract base class. This base 
+class implements the functionality for component initialization, receiving telemetry and event data from devices or 
```

Comment:
The base class only provides methods for forwarding data to the Hono server. It is the `VertxBasedRestProtocolAdapter` class which adds the corresponding routes for receiving data from clients..